mereswine/crawler.py

```diff
@@ -44,9 +48,24 @@ def crawl(instance):
 
 
 def geolocate(instance):
-    url = 'http://freegeoip.net/json/{0}'.format(trim_url(instance.url))
-    r = requests.get(url)
+    api_key = current_app.config.get('IP_GEOLOCATION_API_KEY')
+
+    if not api_key:
+        logging.warn('Geolocation API key not set. Geolocation is off.')
+        return
+
+    url = trim_url(instance.url)
+    r = requests.get(GEOLOCATION_URL, params={
+        'ip': socket.gethostbyname(url),
```

Comment:
This will probably create unhandled exceptions when crawling those invalid instances.

Reply:
Should be fine now. I also replaced that odd `trim_url` function with `urlparse`...